src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKPainter.java

```diff
@@ -612,7 +612,7 @@ public void paintPopupMenuBackground(SynthContext context, Graphics g,
             BufferedImage img = ENGINE.finishPainting();
             if(!isHW) {
                 int border = img.getRGB(0, h / 2);
-                if (img != null && border == img.getRGB(w / 2, h / 2)) {
+                if (border == img.getRGB(w / 2, h / 2)) {
                     // fix no menu borders in Adwaita theme
                     Graphics g2 = img.getGraphics();
```

Comment:
I guess we can do `if (!isHw && img!=null)`
Also the comment that it is done for Adwaita theme, shouldn't then this code block be done in native after checking what theme is effective now, as this may affect Yaru or other theme potentially, no?

Reply:
> I guess we can do if (!isHw && img!=null)

Thought of but "is it possible that img can be null when it is returned from GTKEngine.finishPainting() method as the BufferedImage instance is created there and returned? ".

Reply:
GTKEngine is not final so it is possible for it to be extended and provided implementation to finishPainting so it better to safeguard...also to be noted why only in this method, finishPainting returned image is being used, other several places, it does not check the returned image at all..

Reply:
> GTKEngine is not final so it is possible for it to be extended and provided implementation to finishPainting so it better to safeguard

Ok, got it. Will updated.

> Also the comment that it is done for Adwaita theme, shouldn't then this code block be done in native after checking what theme is effective now, as this may affect Yaru or other theme potentially, no?
 also to be noted why only in this method, finishPainting returned image is being used, other several places, it does not check the returned image at all..

I am checking the test in my local machine (Ubuntu_22.04 which is having Yaru theme) for which the code is added initially. Will update my findings.